### DISCUSSION TITLE
what's the relationship of the activation phase with the rendered HPA spec

### DISCUSSION THREAD

**Discussion by matheuscscp** [kedacore/keda#3645] [Q&A / Need Help]
hi all! we have a couple of use cases for the Prometheus scaler in Arista Networks and would like to clarify a few things, if anybody would be kind enough to help :grin: 

in more recent versions of the `autoscaling` API it's possible to set `minReplicas = 0` for an HPA: so what exactly is the benefit of the activation phase? also, looking at the KEDA code it looked to me that there isn't a significant amount of coordination between the activation and the scaling phases driven by the events/metrics in the sense that the HPA is created when the ScaledObject is created, and then it's updated only if the ScaledObject's rendered HPA `.spec` changes, while the activation phase runs in a separate loop and doesn't touch the HPA spec. so it looks to me that the these two phases are just two overlapping things... I was expecting to see the activation phase have some effect over the HPA spec, in practice it only sets the `replicas` field of the `/scale` subresource (e.g. the `replicas` field of a `Deployment`)

**Comment by ghost** [comment]
OK I think I got it: https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/

> `ScalingActive`, indicates whether or not the HPA is enabled (i.e. the replica count of the target is not zero)

Looks like the historic reason for HPA not allowing zero in `minReplicas` is that it uses the `replicas` field of the target to know whether it should do something or not, i.e. it works like an "activating" gate for the HPA.

I wonder if that changed after `minReplicas` started being allowed to be zero...

**Comment by ghost** [comment]
also it doesn't look like `HPAScaleToZero` is becoming "official" soon: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

**Comment by zroubalik** [comment]
Yeah, you covered it pretty well. HPA and scale to zero is still not mature enough. That's why we need to have this approach inside KEDA.
